src/common/chat-participants/powerpages/PowerPagesChatParticipant.ts

```diff
@@ -159,9 +160,23 @@ export class PowerPagesChatParticipant {
             //TODO: Handle command scenarios
 
         } else {
+            const relatedFiles: IRelatedFiles[] = [];
+
+            // Based on dataverse entity fetch required context for the active file
+            switch (activeFileParams.dataverseEntity) {
+                case 'adx_webpage':
+                    if (activeFileUri) {
+                        const files = await fetchRelatedFiles(activeFileUri, activeFileParams.dataverseEntity, activeFileParams.fieldType, this.telemetry, this.powerPagesAgentSessionId);
+                        relatedFiles.push(...files);
+                    }
+                    break;
+                default:
+                    break;
+            }
+
             const { componentInfo, entityName }: IComponentInfo = await getComponentInfo(this.telemetry, this.orgUrl, activeFileParams, this.powerPagesAgentSessionId);
 
-            const llmResponse = await sendApiRequest([{ displayText: userPrompt, code: activeFileContent }], activeFileParams, this.orgID, intelligenceApiToken, this.powerPagesAgentSessionId, entityName, componentInfo, this.telemetry, intelligenceAPIEndpointInfo.intelligenceEndpoint, intelligenceAPIEndpointInfo.geoName, intelligenceAPIEndpointInfo.crossGeoDataMovementEnabledPPACFlag);
+            const llmResponse = await sendApiRequest([{ displayText: userPrompt, code: activeFileContent }], activeFileParams, this.orgID, intelligenceApiToken, this.powerPagesAgentSessionId, entityName, componentInfo, this.telemetry, intelligenceAPIEndpointInfo.intelligenceEndpoint, intelligenceAPIEndpointInfo.geoName, intelligenceAPIEndpointInfo.crossGeoDataMovementEnabledPPACFlag, relatedFiles);
```

Comment:
we should convert these param into an object now for better readability and param passing

Reply:
Right, have already made that change in next PR.